### PR TITLE
core: minimal windows support.

### PR DESCRIPTION
**Pull request by sergiusens**
Add minimal windows support.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

### CODE REVIEW

File: snapcraft/_options.py [hunk 1]
```diff
@@ -242,7 +243,16 @@ def _set_machine(self, target_deb_arch):
...
+        if sys.platform == 'linux':
+            self.__machine_info = _ARCH_TRANSLATIONS[self.__target_machine]
+        else:
+            self.__machine_info = {
+                'kernel': 'NOBUILD',
+                'deb': 'NOBUILD',
+                'uts_machine': 'NOBUILD',
+                'triplet': 'NOBUILD',
+                'core-dynamic-linker': 'NOBUILD',
+            }
```
Comment (kalikiana) [inline snapcraft/_options.py]: With the placeholders it looks a little cryptic... can you add a comment here to clarify what the code does?

Maybe `# Provider fake machine info to allow container and store commands to work on non-Linux platforms`.

File: setup.py [hunk 2]
```diff
@@ -51,7 +51,12 @@
               'snapcraft.plugins',
               'snapcraft.storeapi'],
     package_data={'snapcraft.internal.repo': ['manifest.txt']},
-    scripts=['bin/snapcraft', 'bin/snapcraft-parser'],
+    entry_points={
+        'console_scripts': [
+            'snapcraft = snapcraft.cli.__main__:run',
+            'snapcraft-parser = snapcraft.internal.parser:main',
+        ],
+    },
```
Comment (kalikiana) [inline setup.py]: I have no idea what difference this makes... can you add a comment? Like `scripts isn't good enough because...`.

Comment (elopio) [inline setup.py]: Can we delete the files in bin now?
Also, they were doing a very ugly thing calling snapcraft.internal.dirs.setup_dirs. Do we no longer need that? I would be very happy if so.

**Comment by elopio** [comment]
If you get this tested in a windows machine, I'm happy.
I left a question about the ugly things that we are doing in bin/snapcraft.

**Comment by sergiusens** [comment]
@elopio this was developed and tested on Windows :wink:

File: setup.py (hunk 2 above)
Comment (sergiusens) [inline setup.py]: @kalikiana this is just standard python, it is a portable way to deliver.
@elopio not ready to delete the scripts in `bin` until we say we don't care about local development anymore.

Comment (kalikiana) [inline setup.py]: If it's standard, why didn't we do it this way before? That's what I mean.

Comment (sergiusens) [inline setup.py]: Oh, because the person that did this originally wasn't a python developer and we never got to change it.

Comment (kyrofa) [inline setup.py]: `scripts` are the old distutils way of doing things, `entry_points` are setuptools. They're the [recommended way of doing this](https://packaging.python.org/tutorials/distributing-packages/#scripts), particularly when you're talking [cross-platform stuff](https://setuptools.readthedocs.io/en/latest/setuptools.html#automatic-script-creation).

**Comment by elopio** [comment]
+1 for me. I'm going to trigger the autopkgtests for xenial, just in case.
And it would be nice to comment why on windows a new pymacaroons is being installed.

**Pull request merged**